### PR TITLE
task-3: added error boundary, components for home page

### PR DESCRIPTION
**Pull request by OlehKhozhai**

### CODE REVIEW

File: src/components/App/index.tsx [hunk 1]
```diff
@@ -1,38 +1,21 @@
...
+        <div className={styles.root}>
+          <MainRouter />
+        </div>
+      </ErrorBoundary>
+    </BrowserRouter>
   );
 };
 
-export default memo(App);
+export default React.memo(App);
```
Comment (yurii-hasiuk) [inline src/components/App/index.tsx:21]: Do we need `memo` here?

File: src/components/App/index.tsx [hunk 2]
```diff
@@ -1,38 +1,21 @@
...
+import ErrorBoundary from 'components/_common/ErrorBoundary';
 
-const CreateElementExample = () => {
-  return createElement('h2', null, 'Create Element Example');
-};
+import MainRouter from 'routing';
+import styles from './styles.module.scss';
 
-const App: FC = () => {
+const App: React.FC = () => {
```
Comment (yurii-hasiuk) [inline src/components/App/index.tsx:9]: Please, use destructuring here. Check all occurrences

File: src/components/_common/Dropdown/index.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  return (
+    <div className={cn(styles.root, className)}>
+      <span className={styles.value} onClick={handleOpenDropdownClick}>
+        {value}
+      </span>
+
+      {isOpen && (
+        <ul className={styles.dropdown}>
+          {options.map((option) => {
+            const handleOptionClick = () => {
```
Comment (yurii-hasiuk) [inline src/components/_common/Dropdown/index.tsx:29]: I would move this function outside of `map`

**Review by yurii-hasiuk — CHANGES_REQUESTED** [review]

File: src/components/App/index.tsx (hunk 2 above)
Comment (OlehKhozhai) [reply src/components/App/index.tsx:9]: usually, I try to refer to this GitHub page with rules for ts+react projects - https://github.com/typescript-cheatsheets/react
image


and another point, that when you test your app with ReactSpy you will need to use React.(any hook) to test hook and it a little bit confusing sometimes use React.useState and in the same file only FC<>, that's why I try to make a common view for those imports.

File: src/components/_common/Dropdown/index.tsx (hunk 3 above)
Comment (OlehKhozhai) [reply src/components/_common/Dropdown/index.tsx:29]: I will move the option to a separate component in the next task

File: src/components/App/index.tsx (hunk 1 above)
Comment (OlehKhozhai) [reply src/components/App/index.tsx:21]: yep, it is not necessary here

**Pull request merged**